### PR TITLE
feat: add TextInputV2 and TextInputFieldV2

### PR DESCRIPTION
**Pull request by fabienhebert**
## Type

- Feature

### Summarise concisely:

Add a new version of `TextInput` and `TextInputFieldV2`

### CODE REVIEW

File: packages/form/src/components/TextInputFieldV2/__stories__/Template.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import type { StoryFn } from '@storybook/react'
+import { Stack } from '@ultraviolet/ui'
+import type { ComponentProps } from 'react'
+import { TextInputField } from '..'
+import { Submit } from '../..'
+
+export const Template: StoryFn<
+  ComponentProps<typeof TextInputField>
+> = args => (
+  <Stack gap="1">
```
Comment (vincentaudebert) [inline packages/form/src/components/TextInputFieldV2/__stories__/Template.stories.tsx:10]:
```suggestion
  <Stack gap={1}>
```

?

File: packages/ui/src/components/TextInputV2/__stories__/Examples.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import { TextInput } from '..'
+import { Link } from '../../Link'
+import { Stack } from '../../Stack'
+import { Text } from '../../Text'
+
+export const Examples: StoryFn<typeof TextInput> = args => {
+  const [value, setValue] = useState<string>('Text')
+
+  return (
+    <Stack gap="2">
```
Comment (vincentaudebert) [inline packages/ui/src/components/TextInputV2/__stories__/Examples.stories.tsx:12]:
```suggestion
    <Stack gap={2}>
```

?

**Comment by vincentaudebert** [comment]
Screenshot 2024-01-08 at 10 35 09

it seems magic wand doesn't work in here?

**Comment by matthprost** [comment]
> Screenshot 2024-01-08 at 10 35 09
> it seems magic wand doesn't work in here?

This is because magic wand needs a function to be working using prop `onRandomize`, I will add it into the story so it actually works

File: packages/form/src/components/TextInputFieldV2/__stories__/Template.stories.tsx (hunk 1 above)
Comment (Loriick) [reply packages/form/src/components/TextInputFieldV2/__stories__/Template.stories.tsx:10]: @vincentaudebert Look like gap can be a `number` or a key of space in the `lightTheme`

**Review by Loriick — APPROVED** [review]

File: packages/ui/src/components/TextInputV2/index.tsx [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+  helper?: ReactNode
+  iconName?: ComponentProps<typeof Icon>['name']
+  id?: string
+  label: string
+  labelDescription?: ReactNode
+  loading?: boolean
+  minLength?: number
+  maxLength?: number
+  name?: string
+  onBlur?: DOMAttributes<HTMLInputElement>['onBlur']
```
Comment (DorianMaliszewski) [inline packages/ui/src/components/TextInputV2/index.tsx:134]:
```suggestion
  onBlur?: DOMAttributes<HTMLTextAreaElement>['onBlur']
```

?

File: packages/ui/src/components/TextArea/index.tsx [hunk 4]
```diff
@@ -22,13 +22,12 @@ const StyledTextAreaAbsoluteStack = styled(Stack)`
...
 type StyledTextAreaProps = {
-  isSuccess: boolean
-  isError: boolean
+  hasSentimentIcon: boolean
   isClearable: boolean
 }
 const StyledTextArea = styled('textarea', {
   shouldForwardProp: prop =>
-    !['isSuccess', 'isError', 'isClearable'].includes(prop),
+    !['hasSentimentIcon', 'isClearable'].includes(prop),
```
Comment (DorianMaliszewski) [inline packages/ui/src/components/TextArea/index.tsx:30]: May be use data attirbutes to don't recompute style ?

**Review by DorianMaliszewski — APPROVED** [review]

File: packages/form/src/components/TextInputFieldV2/__stories__/Template.stories.tsx (hunk 1 above)
Comment (matthprost) [reply packages/form/src/components/TextInputFieldV2/__stories__/Template.stories.tsx:10]: @vincentaudebert we don't have restrictions onto that both are being used

File: packages/ui/src/components/TextInputV2/index.tsx (hunk 3 above)
Comment (matthprost) [reply packages/ui/src/components/TextInputV2/index.tsx:134]: @DorianMaliszewski text input is indeed an HTML input element, we have a TextArea component which is different. TextInput doesn't support text area 😄

**Comment by mhervouet** [comment]
On read-only state, better to not be able to focus on the field (same for hover) 
Capture d’écran 2024-01-10 à 15 03 56

**Comment by matthprost** [comment]
> On read-only state, better to not be able to focus on the field (same for hover) Capture d’écran 2024-01-10 à 15 03 56

The native behavior allows focus on readonly state but not hovering, do we want to sync onto that or remove any effect as you said? Cf: https://developer.mozilla.org/fr/docs/Web/HTML/Attributes/readonly#r%C3%A9sultat

**Comment by mhervouet** [comment]
> > On read-only state, better to not be able to focus on the field (same for hover) Capture d’écran 2024-01-10 à 15 03 56
> 
> The native behavior allows focus on readonly state but not hovering, do we want to sync onto that or remove any effect as you said? Cf: https://developer.mozilla.org/fr/docs/Web/HTML/Attributes/readonly#r%C3%A9sultat

Ok to keep this behavior then it will be easier to identify read-only than disabled. Everything is good then :)

**Review by mhervouet — APPROVED** [review]

**Pull request merged**